python/dolfinx/fem/linearvariationalsolver.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2020 Jørgen S. Dokken
+#
+# This file is part of DOLFINX (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+import typing
+
+from petsc4py import PETSc
+from ufl.equation import Equation
+from dolfinx import fem
+
+
+class LinearVariationalSolver():
+    assemble_b = True
+
+    def __init__(self, variational_problem: Equation, u: fem.Function,
+                 bcs: typing.List[fem.DirichletBC] = [], form_compiler_parameters={}):
+        """
+        Initialize the linear variational solver. By initialization we mean creating the underlying PETSc solver,
+        and the linear algebra structures (the matrix and vector), that can be reused for multiple solve calls.
+
+        .. code-block:: python
+
+            solver = LinearVariationalSolver(a==L, u, [bc0, bc1], form_compiler_parameters={"optimize": True})
```

Comment:
`{"optimize": True}` is some very old legacy, doesn't exist anymore. Please pass `form_compiler_parameters` and `jit_parameters` as well.